### DISCUSSION TITLE
How to I prevent the index page from having the H1 heading "index"?

### DISCUSSION THREAD

**Discussion by blacklightpy** [obsidian-html/obsidian-html#764] [Q&A] [answered]
I wanted my entrypoint file to retain it's original filename as the heading, but if I don't specify an H1 heading, it always gets the default "index" heading. Other notes don't have this problem.

As a workaround, I added an H1 heading, so with the default config it used the H1 heading instead of the filename as the first heading.

But then later I saw the `hide_on_h1`, and I wanted to keep it `false` so every note gets a title. But doing that again gives the index note the "index" title. 

```yaml
embedded_note_titles:
      enabled: true
      hide_on_h1: false
```
I went through the source code a lot, but I couldn't figure out what was happening. I'd guess while the file is renamed to index.md during markdown generation the embedded note title is obtained from the markdown filenames, leaving index.md titled as "index"

I couldn't understand the generation and title insertion steps well enough to make it preserve the original filename of the index file.. what can I do?

**Comment by dwrolvink** [comment, marked as answer]
> I went through the source code a lot, but I couldn't figure out what was happening. I'd guess while the file is renamed to index.md during markdown generation the embedded note title is obtained from the markdown filenames, leaving index.md titled as "index"

Yes, that is exactly what is happening.  

I'd suggest you keep the embedded_note_titles settings that you posted, and then add the following to the frontmatter yaml block in your entrypoint:
``` yaml
obs.html.tags:
- dont_add_embedded_title
```
> https://obsidian-html.github.io/configurations/note-settings/note-settings.html#!dont_add_embedded_title

So that you can have your embedded titles everywhere except for your index page. I think that would be the easiest solution.

Turns out I did the same for the documentation website, you can check out the entrypoint note for that here:
- https://github.com/obsidian-html/obsidian-html.github.io/blob/main/__src/vault_v4/ObsidianHtml.md?plain=1
